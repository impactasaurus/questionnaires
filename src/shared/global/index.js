import { injectGlobal } from 'styled-components';
import styledReset from 'styled-reset';

export default () => injectGlobal`
    @import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,300italic,400,400italic");
    ${styledReset}

    * {
        box-sizing: border-box;
    }
`;