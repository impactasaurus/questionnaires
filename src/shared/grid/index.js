import R from './Row';
import C from './Column';

export const Row = R;
export const Column = C;