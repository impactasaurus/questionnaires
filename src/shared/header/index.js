import Header from './Header';
import Sub from './Subheader';

export const Subheader = Sub;

export default Header;