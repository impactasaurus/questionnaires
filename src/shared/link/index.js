import Link, { BlankLink as RawBlankLink } from './Link';

export const BlankLink = RawBlankLink;
export default Link;