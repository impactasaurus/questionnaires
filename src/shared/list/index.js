import List, {ListItem as LI} from './List';

export const ListItem = LI;
export default List;