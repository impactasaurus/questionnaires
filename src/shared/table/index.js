import Table from './Table';
import Header from './TableHeader';
import Body from './TableBody';
import VTable from './VerticalTable';

export const TableHeader = Header;
export const TableBody = Body;
export const VerticalTable = VTable;
export default Table;