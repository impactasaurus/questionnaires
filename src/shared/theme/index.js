import rawTypeset from './typeset';
import rawColours from './colours';
import rawSpacings from './spacings';

export const typeset = rawTypeset
export const colours = rawColours;
export const spacings = rawSpacings;

export const primaryTheme = {
    fg: colours.gray60,
    bg: 'transparent',
    highlight: colours.green50,
    lowlight: colours.white,
    gradient: colours.primaryGradient,
};

export const secondaryTheme = {
    fg: colours.white,
    bg: colours.blue30,
    highlight: colours.white,
    lowlight: 'transparent',
    gradient: colours.primaryGradient,
};