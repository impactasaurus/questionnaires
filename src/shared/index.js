import RawAction from './actions';
import RawButton from './button';
import RawHeader, { Subheader as Subhead } from './header';
import RawHero from './hero';
import RawLink, {BlankLink as BLink } from './link';
import RawSection from './section';
import RawTable, { VerticalTable as VTable, TableHeader as TH, TableBody as TB } from './table';
import RawList, { ListItem as LI } from './list';
import { Row as R, Column as C } from './grid';
import RawText from './text';
import RawLogo from './logo';

export const Action = RawAction;
export const Button = RawButton;
export const Header = RawHeader;
export const Subheader = Subhead;
export const Hero = RawHero;
export const Link = RawLink;
export const BlankLink = BLink;
export const Section = RawSection;
export const Table = RawTable;
export const TableHeader = TH;
export const TableBody = TB;
export const VerticalTable = VTable;
export const List = RawList;
export const ListItem = LI;
export const Row = R;
export const Column = C;
export const Text = RawText;
export const Logo = RawLogo;