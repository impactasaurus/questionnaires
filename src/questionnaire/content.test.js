import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { StaticRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { primaryTheme } from '../shared/theme';
import { Hero, Table, VerticalTable } from '../shared';
import ConnectedContent, { Content } from './Content';

describe('questionnaire <Content />', () => {
    const render = (component, initialState) => {
        const store = createStore([ thunk ])(initialState || {});
        return renderer.create(
            <Provider store={store}>
                <StaticRouter context={{}}>
                    <ThemeProvider theme={primaryTheme}>
                        {component}
                    </ThemeProvider>
                </StaticRouter>
            </Provider>
        );
    };

    const data = {
        id: '1',
        name: 'Questionniare name',
        description: 'Questionnaire description',
        pdfLink: 'questionnaire.link',
        sectors: ['some', 'sectors'],
        demographic: ['some', 'demographic'],
        scoring: {
            aggregation: 'sum',
            bands: [
                { minimum: 1, maximum: 5, label: 'Low risk' },
                { minimum: 6, maximum: 10, label: 'Medium risk' },
                { minimum: 11, maximum: 20, label: 'High risk' },
            ],
        },
        questions: [
            {
                id: '1',
                text: 'Question text',
                left: {
                    label: 'Strongly disagree',
                    score: 1,
                },
                right: {
                    label: 'Strongly agree',
                    score: 5,
                },
                categories: ['some', 'categories'],
            },
        ],
    };

    it('should load the questionnaire from the api', () => {
        const loadQuestionnaireSpy = jest.fn();
        render(<Content loadQuestionnaire={loadQuestionnaireSpy} />);
        expect(loadQuestionnaireSpy).toHaveBeenCalled();
    });

    it('should pass the data to the tables', () => {
        const component = render(<Content loadQuestionnaire={jest.fn()} data={data} />).root;
        const summaryTable = component.findByType(VerticalTable);
        expect(summaryTable.props.data).toEqual([data]);
        const mainTable = component.findByType(Table);
        expect(mainTable.props.data).toEqual(data.questions);
    });

    it('should match the snapshot', () => {
        const component = render(<Content loadQuestionnaire={jest.fn()} data={data} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});