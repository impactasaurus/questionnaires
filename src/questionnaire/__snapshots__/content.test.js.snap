// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`questionnaire <Content /> should match the snapshot 1`] = `
.c3 {
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  color: rgb(153,153,153);
  line-height: 3.375rem;
  font-size: 2.25rem;
  font-weight: 300;
  margin: 0;
}

.c3::after {
  content: '';
  height: 1px;
  display: block;
  width: 8rem;
  margin-top: 0.75rem;
  border-bottom: 1px solid rgb(153,153,153);
}

.c11 {
  color: rgb(153,153,153);
  border-bottom: 1px dotted rgb(153,153,153);
  text-decoration: none;
  -webkit-transition: color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  line-height: 1.5rem;
  font-size: 1rem;
}

.c11:hover {
  border-bottom-color: transparent;
  color: rgb(157,198,107);
}

.c0 {
  padding: 1.5rem 2.25rem;
}

.c16 {
  border-bottom: 1px solid rgb(153,153,153);
}

.c17:nth-child(2n) {
  background-color: rgba(157,198,107,0.075);
}

.c15 {
  width: 100%;
}

.c9 {
  color: rgb(153,153,153);
  text-align: left;
  padding: 0 0.75rem 0.75rem 0.75rem;
  font-family: "Source Sans Pro",Helvetica,sans-serif;
}

.c10 {
  padding: 0.75rem;
}

.c8:nth-child(2n + 1) {
  background-color: rgba(157,198,107,0.075);
}

.c7 {
  width: 100%;
}

.c14 {
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 300;
  color: rgb(153,153,153);
  padding: 0.5em 0;
}

.c14:first-of-type {
  padding-top: 0;
}

.c14:last-of-type {
  padding-bottom: 0;
}

.c12 {
  list-style: none;
}

.c12 .c13 {
  border-top: 1px solid rgb(153,153,153);
}

.c12 .c13:first-of-type {
  border-top: 0;
  padding-top: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-bottom: 1.5rem;
}

.c1 > * {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c1::after {
  content: '';
  display: table;
  clear: both;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-bottom: 2.625rem;
}

.c5 > * {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c5::after {
  content: '';
  display: table;
  clear: both;
}

.c2 {
  width: 100%;
}

.c6 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  color: rgb(153,153,153);
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  font-size: 1rem;
  font-weight: 300;
  line-height: 1.5rem;
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h1
        className="c3"
      >
        Questionniare name
      </h1>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <p
        className="c4"
      >
        Questionnaire description
      </p>
    </div>
  </div>
  <div
    className="c5"
  >
    <div
      className="c6"
    >
      <table
        className="c7"
      >
        <tbody>
          <tr
            className="c8"
          >
            <th
              className="c9"
            >
              PDF Link
            </th>
            <td
              className="c10"
            >
              <a
                className="c11"
                href="/questionnaire.link"
                onClick={[Function]}
                target="_blank"
              >
                questionnaire.link
              </a>
            </td>
          </tr>
          <tr
            className="c8"
          >
            <th
              className="c9"
            >
              Sectors
            </th>
            <td
              className="c10"
            >
              <p
                className="c4"
              >
                some, sectors
              </p>
            </td>
          </tr>
          <tr
            className="c8"
          >
            <th
              className="c9"
            >
              Demographic
            </th>
            <td
              className="c10"
            >
              <p
                className="c4"
              >
                some, demographic
              </p>
            </td>
          </tr>
          <tr
            className="c8"
          >
            <th
              className="c9"
            >
              Scoring
            </th>
            <td
              className="c10"
            >
              <p
                className="c4"
              >
                sum
              </p>
            </td>
          </tr>
          <tr
            className="c8"
          >
            <th
              className="c9"
            >
              Scoring bands
            </th>
            <td
              className="c10"
            >
              <ul
                className="c12"
              >
                <li
                  className="c13 c14"
                >
                  1
                   - 
                  5
                  : 
                  Low risk
                </li>
                <li
                  className="c13 c14"
                >
                  6
                   - 
                  10
                  : 
                  Medium risk
                </li>
                <li
                  className="c13 c14"
                >
                  11
                   - 
                  20
                  : 
                  High risk
                </li>
              </ul>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <table
        className="c15"
      >
        <thead
          className="c16"
        >
          <tr>
            <th
              className="c9"
            >
              Question
            </th>
            <th
              className="c9"
            >
              Left score
            </th>
            <th
              className="c9"
            >
              Right score
            </th>
            <th
              className="c9"
            >
              Categories
            </th>
          </tr>
        </thead>
        <tbody
          className=""
        >
          <tr
            className="c17"
          >
            <td
              className="c10"
            >
              <p
                className="c4"
              >
                Question text
              </p>
            </td>
            <td
              className="c10"
            >
              <p
                className="c4"
              >
                Strongly disagree
                 (
                1
                )
              </p>
            </td>
            <td
              className="c10"
            >
              <p
                className="c4"
              >
                Strongly agree
                 (
                5
                )
              </p>
            </td>
            <td
              className="c10"
            >
              <p
                className="c4"
              >
                some, categories
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</section>
`;
